fnmatch/fnmatch.py

```diff
@@ -50,9 +59,15 @@ def filter(names, pat):
     result = []
     pat = os.path.normcase(pat)
     match = _compile_pattern(pat)
-    for name in names:
-        if match(os.path.normcase(name)):
-            result.append(name)
+    if os.path is posixpath:
+        # normcase on posix is NOP. Optimize it away from the loop.
```

Comment:
Have you measured if this has an actual impact?

Reply:
I didn't specifically make this change either. Wonder what is going on.